### PR TITLE
library: Change cover art resolution order.

### LINKED ISSUE #386: Missing Image for JayDLC / The messenger picnic panic
**Describe the bug**
#81 was closed because there was no image data available. In this case the image data is available but Heroic still says "Image not found":

image

**To Reproduce**

1. Login to an account that has "The messenger Picnic Panic"
2. View the image in Library

**Expected behavior**
The background loads, one of 

```json
    "keyImages": [
      {
        "height": 1440,
        "md5": "0466026af367640bd29a5052f326b1e7",
        "size": 541703,
        "type": "DieselGameBox",
        "uploadedDate": "2019-11-13T18:34:35.337Z",
        "url": "https://cdn1.epicgames.com/752368fdba244e9386023cfb2b9d6ceb/item/Sabotage_PicnicPanic-2560x1440-0466026af367640bd29a5052f326b1e7.jpg",
        "width": 2560
      },
      {
        "height": 420,
        "md5": "6b715bcdcd8d7ace25ec786ec82dda6a",
        "size": 75843,
        "type": "DieselGameBoxLogo",
        "uploadedDate": "2019-11-13T18:35:03.531Z",
        "url": "https://cdn1.epicgames.com/752368fdba244e9386023cfb2b9d6ceb/item/ColorPicnicPanicLogo-986x420-6b715bcdcd8d7ace25ec786ec82dda6a.png",
        "width": 986
      },
      {
        "height": 1440,
        "md5": "f1232fd3929b62f0f6e0eecce04120c4",
        "size": 268204,
        "type": "DieselStoreFrontTall",
        "uploadedDate": "2019-11-13T18:35:12.463Z",
        "url": "https://cdn1.epicgames.com/752368fdba244e9386023cfb2b9d6ceb/item/PicnicPanic_Portrait-1280x1440-f1232fd3929b62f0f6e0eecce04120c4.jpg",
        "width": 1280
      }
    ],
```

**Screenshots**
If you look closely, you can see that the custom text has actually loaded:
image

**Desktop (please complete the following information):**

- OS: Manjaro, 
- Version: Heroic 1.7.0-1
- Game: 
 -  "app_name": "JayDLC1",
 -  "app_title": "The Messenger Picnic Panic",
- Wine/Proton version: 6.8

**Additional context**

### PR DESCRIPTION
**Pull request by StefanLobbenmeier**
New order of resolution for art_square: art_square -> art_cover -> fallBackImage

Should fix #386 where the image "DieselGameBoxTall" was missing in the JSON

### CODE REVIEW

**Comment by StefanLobbenmeier** [comment]
This is what the cover now looks like on my machine:
image

**Comment by flavioislima** [comment]
Ah OK, you already got the fix. Nice.

**Review by flavioislima — CHANGES_REQUESTED** [review]
I think that instead of using the cover for the fallback, use the dieselStoreFrontTall type that comes with this game and probably others and then the cover. So it will look better.
So would be dieselGameBoxTall, if empty, DieselStoreFrontTall and last DieselGameBox.

👍

**Comment by flavioislima** [comment]
@StefanLobbenmeier do you want to fix it or should we fix it?

**Comment by StefanLobbenmeier** [comment]
Oh sorry hadn’t seen the first response. It’s probably less work for everyone if you fix it yourself since you already know exactly what you want :) 

I volunteer to get another screenshot after your path though

**Review by adityaruplaha — DISMISSED** [review]

**Comment by flavioislima** [comment]
@StefanLobbenmeier Can you confirm is working after my changes?

**Comment by StefanLobbenmeier** [comment]
Thanks! Just tried it on MacOS, but the game list does not load. I will try again on Sunday on Linux

**Comment by StefanLobbenmeier** [comment]
Ok fixed my issue for MacOS, the logs said
`/bin/sh: /Applications/Heroic: No such file or directory`
so I renamed the application from `Heroic Games Launcher` to `HeroicGamesLauncher` and the issue disappeared. Will open a separate issue for that.

Anyway after your changes the cover looks the same as with my approach, so it looks good to me:
image

**Comment by flavioislima** [comment]
> Ok fixed my issue for MacOS, the logs said
> `/bin/sh: /Applications/Heroic: No such file or directory`
> so I renamed the application from `Heroic Games Launcher` to `HeroicGamesLauncher` and the issue disappeared. Will open a separate issue for that.
> 
> Anyway after your changes the cover looks the same as with my approach, so it looks good to me:
> image
> 

Really? It should have worked with the change. Até you sure you're on the latest commit?

The mac version is complicated because we don't have a dedicated maintener so I don't even know if we will launch a installer for it. There are several things that needs to be done for it work as well.

**Comment by StefanLobbenmeier** [comment]
I am on commit 521620abae366b7d3725b07b94e100b275da25f9, made a fresh checkout on another machine for this

**Comment by StefanLobbenmeier** [comment]
```
➜  HeroicGamesLauncher git:(main) git log



commit 521620abae366b7d3725b07b94e100b275da25f9 (HEAD -> main, origin/main, orig
in/HEAD)
Merge: f77a4f2 6a35318
Author: Flávio F Lima <flavioislima@gmail.com>
Date:   Thu Jun 3 20:22:48 2021 +0200

    Merge branch 'main' into main

commit f77a4f268207b3bc6c24e975cce3920b84a73f67
Author: Flávio F Lima <flavioislima@gmail.com>
Date:   Thu Jun 3 20:20:17 2021 +0200

    chore: lint

commit 1f297b4eb3a5041a7bd7bc4af09b95d81d1b91c8
Author: Flávio F Lima <flavioislima@gmail.com>
Date:   Thu Jun 3 20:15:19 2021 +0200

    feat: added dieselStoreFrontTall as another fallback
```

**Comment by StefanLobbenmeier** [comment]
I just checked your implementation via some debug logs:

```
        const artSquareResolved = art_square || art_square_front || art_cover || fallBackImage;
        console.log(app_name, art_square);
        console.log(app_name, art_square_front);
        console.log(app_name, art_cover);
        console.log(app_name, artSquareResolved);
```
```
JayDLC1 null
library.js:273JayDLC1 null
library.js:274JayDLC1 https://cdn1.epicgames.com/752368fdba244e9386023cfb2b9d6ceb/item/Sabotage_PicnicPanic-2560x1440-0466026af367640bd29a5052f326b1e7.jpg
library.js:275JayDLC1 https://cdn1.epicgames.com/752368fdba244e9386023cfb2b9d6ceb/item/Sabotage_PicnicPanic-2560x1440-0466026af367640bd29a5052f326b1e7.jpg
```

File: electron/legendary_utils/library.ts [hunk 1]
```diff
@@ -272,6 +272,10 @@ class Library {
     const gameBoxTall = is_game ?
       keyImages.filter(({ type }: KeyImage) => type === 'DieselGameBoxTall' )[0] :
       gameBox
+    
+    const gameBoxStore = is_game ?
+      keyImages.filter(({ type }: KeyImage) => type === 'dieselStoreFrontTall' )[0] :
```
Comment (StefanLobbenmeier) [inline electron/legendary_utils/library.ts:300]: In the JSON in #386 dieselStoreFrontTall is actually uppercase DieselStoreFrontTall

**Comment by StefanLobbenmeier** [comment]
After settings the String to uppercase I get another image now:

**Comment by flavioislima** [comment]
Perfect, good catch!
I'll merge it then

**Review by flavioislima — APPROVED** [review]

**Pull request merged**